bundles/org.openhab.binding.systeminfo/src/main/java/org/openhab/binding/systeminfo/internal/handler/SysteminfoHandler.java

```diff
@@ -241,10 +241,14 @@ private boolean addDynamicChannels() {
             newChannelGroups.addAll(0, thingTypeProvider.getChannelGroupDefinitions(thing.getThingTypeUID()));
             ThingTypeUID thingTypeUID = new ThingTypeUID(BINDING_ID, THING_TYPE_COMPUTER_ID + idExtString);
             if (thingTypeProvider.updateThingType(thingTypeUID, newChannelGroups)) {
-                logger.trace("Channel groups where added, changing the thing type");
+                logger.trace("Channel groups were added, changing the thing type");
                 changeThingType(thingTypeUID, thing.getConfiguration());
+                return true;
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                        "@text/offline.cannot-initialize");
+                return false;
```

Comment:
If you return false, this will trigger the additional code in `initialize` that finally sets the status to ONLINE.
So either you have to return true there or you return false but do not set the status to OFFLINE there.
Something looks not coherent.

Reply:
@lolodomo Indeed, didn't think it through well enough. I think the new commit should fix that.